bloom-inference-scripts/bloom-ds-inference.py

```diff
@@ -227,7 +228,7 @@ def write_checkponts_json():
     # dynamically extend to support larger bs by repetition
     input_sentences *= math.ceil(args.batch_size / len(input_sentences))
 
-generate_kwargs = dict(max_new_tokens=num_tokens, do_sample=False)
+generate_kwargs = dict(max_new_tokens=num_tokens, do_sample=True)
```

Comment:
this is already a very different type of change. 

If `int4` requires `do_sample=True`, then again, let's change it only if it's `--dtype int4`

Reply:
Sure, I will double check the `do_sample=False` again to see if the text generated makes sense. If not, I set it to true for int4

Reply:
I just checked with `do_sample=False` and I see the text is produced in the same way as for FP16 and INT8:
```
in=DeepSpeed is a machine learning framework
out=DeepSpeed is a machine learning framework for deep learning. It is a Python library, and it is also a framework. It is a framework, and it is a library. It is a framework, and it is a library. It is a framework, and it is a library. It is a framework, and it is a library. It is a framework, and it is a library. It is a framework, and it is a library. It is a framework, and it is a library. It is a framework, and
```
So, I am gonna turn it off for now.